### ISSUE TITLE
All messages show encrypted, Impossible to communicate with each other or transact.

### ISSUE THREAD

**Issue by nostitos** [nunchuk-io/libnunchuk#25]
It's hard to describe, Simply nothing works anymore.

We're logged with primary keys.

**Comment by plebhash** [comment]
any updates here?

facing similar issues with @diegokolling

[according to](https://stacker.news/items/219635?commentId=219790) @hugohn it's related to Matrix SDK?

**Comment by hugohn** [comment]
Correct, the error is related to the Matrix SDK.

Current solution is to recover the wallet manually. I'm sharing the steps below.

**How to manually recovering a Matrix-backed collaborative wallet**
1. Back up your keys and wallets (BSMS), if you have not.
2. Delete the chat, and the associated wallet.
3. Have all co-signers update the app.
4. Have all co-signers recover their respective key in the app.
5. Create a new group chat and invite the original members.
6. Once all original members have joined (and only then), recover the original wallet in the new chat by doing "Create a new collaborative wallet" > "Recover wallet" > Select BSMS file from step 1.

**Comment by diegokolling** [comment]
Hey, Hugo! Good to have an answer from you.

The point is we can’t even create the collaborative wallet, since we can
interact through the chat, it seems. I can’t see Plebhashs messages of any
form.

Enviado do Gmail para celular


Em sáb., 27 de jul. de 2024 às 04:58, Hugo ***@***.***>
escreveu:

**Comment by plebhash** [comment]
on a side note: I used the `Element` client (which is built on the `Matrix` protocol) on a regular basis until not long ago.

this would always happen while bootstrapping new devices on the same account.

---

> Correct, the error is related to the Matrix SDK.
> 
> Current solution is to recover the wallet manually. I'm sharing the steps below.
> 
> **How to manually recovering a Matrix-backed collaborative wallet**
> 
> 1. Back up your keys and wallets (BSMS), if you have not.
> 2. Delete the chat, and the associated wallet.
> 3. Have all co-signers update the app.
> 4. Have all co-signers recover their respective key in the app.
> 5. Create a new group chat and invite the original members.
> 6. Once all original members have joined (and only then), recover the original wallet in the new chat by doing "Create a new collaborative wallet" > "Recover wallet" > Select BSMS file from step 1.

will these steps also work for our case?

I believe @diegokolling already had the `nunchuk` app on this mobile for a while. I on the other hand had just installed it, so I think our case sort of falls into the "bootstrapping" category as well (similar to what I described about my experience with `Element` above)

**Comment by hugohn** [comment]
@diegokolling you shouldn’t get the Encrypted messages if you do a fresh install. However all members must do it.

**Comment by diegokolling** [comment]
Hey Hugo! Yeah, the problem is that reinstalling the wallet would be a
stressful event due to the sensitive type of data to recover. I saw
somewhere that you are planning some kind of authentication that would fix
this kind of problem, no?

It seems to be the weakest link in the chain of the collaborative wallet.

Enviado do Gmail para celular


Em sáb., 27 de jul. de 2024 às 23:46, Hugo ***@***.***>
escreveu:

**Comment by hugohn** [comment]
No, I’m not sure what you meant by “some kind of authentication”?

If you don’t do a fresh install, you’ll be stuck with the old set of Matrix E2EE keys, so messages could not be decrypted. This is unfortunately the architecture of the free collaborative wallet which currently relies on the Matrix SDK.

A fresh install would allow all parties to reset their key sets and start over. As long as you have the original keys and wallet config backed up, you should always be able to recover the same collaborative wallet within a new chat.

For a more reliable version of collaborative wallet, I suggest the Finney service.

**Comment by diegokolling** [comment]
When installing Nunchuk for a family member recently, I saw somewhere in
the chat that, in a future update, you would authenticate something that
would allow to chat using multiple devices or something like that.

I though that would fix our problem.

But as it seems, it’s unfixable if I do not reinstall the app, right?
There’s no hope in any future updates If I understood it correctly.

Enviado do Gmail para celular


Em dom., 28 de jul. de 2024 às 09:18, Hugo ***@***.***>
escreveu:

**Comment by Cyberpunk-Hardware** [comment]
Can you please confirm that Finney doesn't use matrix and what it uses?
